src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerPathResolver.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Collections.Immutable;
+using System.Globalization;
+
+namespace Microsoft.CodeAnalysis.Remote.Diagnostics;
+
+/// <summary>
+/// For analyzers shipped in Roslyn, different set of assemblies might be used when running
+/// in-proc and OOP e.g. in-proc (VS) running on desktop clr and OOP running on ServiceHub .Net6
+/// host. We need to make sure to use the ones from the same location as the remote.
+/// </summary>
+internal sealed class RemoteAnalyzerPathResolver(string baseDirectory) : IAnalyzerPathResolver
+{
+    private readonly string _baseDirectory = baseDirectory;
+
+    private string GetFixedPath(string analyzerPath)
+        => Path.GetFullPath(Path.Combine(_baseDirectory, Path.GetFileName(analyzerPath)));
+
+    public bool IsAnalyzerPathHandled(string analzyerPath)
+        => File.Exists(GetFixedPath(analzyerPath));
```

Comment:
can we doc this.  i don't quite understand this.  this feels racey/weird to me.  specifically that it oculd return *different* values depending on if the copy happeend or not...

Reply:
What is racy here? Either the path exists or it doesn't. There is a hard assumption, fully documented, that the file system is considered unchanged while the analyzer loader is executing.